src/Nri/Ui/SlideModal/V1.elm

```diff
@@ -0,0 +1,308 @@
+-}
+type alias Panel msg =
+    { icon : Html Never
+    , title : String
+    , content : Html msg
+    , buttonLabel : String
+    }
+
+
+viewPanels : (State -> msg) -> Summary msg -> ( String, List (Html msg) )
+viewPanels parentMsg ({ current } as summary) =
+    let
+        id =
+            "modal-header__" ++ String.replace " " "-" current.title
+    in
+    ( id
+    , [ viewIcon current.icon
+      , Text.subHeading
+            [ span [ Html.Styled.Attributes.id id ] [ Html.text current.title ]
+            ]
+      , viewContent current.content
+      , viewFooter summary |> Html.map parentMsg
+      ]
+    )
+
+
+viewContent : Html msg -> Html msg
+viewContent content =
+    Nri.Ui.styled div
+        "modal-content"
+        [ Css.overflowY Css.auto
+        , Css.padding2 (Css.px 30) (Css.px 45)
+        , Css.width (Css.pct 100)
+        , Css.minHeight (Css.px 150)
+        , Css.marginBottom Css.auto
+        , Css.boxSizing Css.borderBox
+        ]
+        []
+        [ content ]
+
+
+viewIcon : Html Never -> Html msg
+viewIcon svg =
+    div
+        [ css
+            [ Css.width (Css.px 100)
+            , Css.height (Css.px 100)
+            , Css.displayFlex
+            , Css.alignItems Css.center
+            , Css.justifyContent Css.center
+            , Css.Global.children
+                [ Css.Global.svg
+                    [ Css.maxHeight (Css.px 100)
+                    , Css.width (Css.px 100)
+                    ]
+                ]
+            ]
+        ]
+        [ svg ]
+        |> Html.map never
+
+
+viewFooter : Summary msg -> Html State
+viewFooter { previous, current, upcoming } =
+    let
+        nextPanel =
+            List.head upcoming
+                |> Maybe.map Tuple.first
+                |> Maybe.withDefault closed
+    in
+    Nri.Ui.styled div
+        "modal-footer"
+        [ Css.displayFlex
+        , Css.flexDirection Css.column
+        , Css.alignItems Css.center
+        , Css.margin4 (Css.px 20) Css.zero Css.zero Css.zero
+        ]
+        []
+        [ viewFooterButton { label = current.buttonLabel, msg = nextPanel }
+        , (List.map (uncurry Inactive) previous
+            ++ Active
+            :: List.map (uncurry InactiveDisabled) upcoming
+          )
+            |> List.map dot
+            |> div [ css [ Css.marginTop (Css.px 16) ] ]
+        ]
+
+
+uncurry : (a -> b -> c) -> ( a, b ) -> c
+uncurry f ( a, b ) =
+    f a b
+
+
+viewFooterButton : { label : String, msg : msg } -> Html msg
+viewFooterButton { label, msg } =
+    Button.button
+        { onClick = msg
+        , size = Button.Large
+        , style = Button.Primary
+        , width = Button.WidthExact 230
+        }
+        { label = label
+        , state = Button.Enabled
+        , icon = Nothing
+        }
+
+
+type Dot
+    = Active
+    | Inactive State String
+    | InactiveDisabled State String
+
+
+dot : Dot -> Html.Html State
+dot type_ =
```

Comment:
🤔 We use these dots on various places... should we add a PT story/chore to make a separate doodad for these?

Reply:
I'll add a tech debt card, if that works?

Reply:
That would be great. Thank you!